### DISCUSSION TITLE
Question about git dependencies

### DISCUSSION THREAD

**Discussion by brianhuster** [lumen-oss/rocks.nvim#596] [Q&A]
As a plugin author, I just want to know if this plugin can handle Git dependencies. Thank you

**Comment by mrcjkb** [comment]
Yes, we have a rocks-git.nvim extension for installing plugins using git.
There's also rocks-dev.nvim that plugin developers use to add plugins from the local filesystem.

**Comment by brianhuster** [comment]
No, I mean it would be nice if I can declare a Git dependency in rockspec and rocks-git.nvim can autodownload it. Sometimes that dependency maybe in Vimscript

**Reply by mrcjkb** [reply]
You can, but the git dependency has to have a rockspec that is published to luarocks.org as a `dev` or `scm` version.
If it isn't, you can create a rockspec for the git dependency and publish it (you don't have to be the package maintainer to do that).

However, if you depend on a git dependency, your package should also be an `scm` version and not SemVer, because you cannot make any stability guarantees.

**Reply by brianhuster** [reply]
So I can publish a Vimscript plugin to LuaRocks?

**Reply by mrcjkb** [reply]
https://luarocks.org/modules/neorocks/vim-test/2.1.0-1 😉

**Reply by brianhuster** [reply]
Ok, but I won't upload a plugin that isn't mine to Luarocks, personally I don't want my plugin to be published to somewhere without my permission. Maybe you can can consider this a feature request, like making rockspec supports Git repo as well (AFAIK you are working on an improved Luarocks client, right)?

**Reply by mrcjkb** [reply]
>  I don't want my plugin to be published to somewhere without my permission

A Rockspec is just a file that tells luarocks how to download and build a package. 
The only way to "forbid" someone from publishing a rockspec for your plugin would be by hosting the source code in a private repository and forbidding redistribution via a proprietary license. 
We publish a bunch of plugins via nvim-neorocks/nurr. If package authors decide they want to take over ownership of the published rockspec, we're more than happy to transfer it to them.
If someone were to be angered by that, they should reconsider whether they should be publishing open source software.

> Maybe you can can consider this a feature request, like making rockspec supports Git repo as well (AFAIK you are working on an improved Luarocks client, right)?

https://github.com/nvim-neorocks/rocks.nvim/issues/539#issuecomment-2495558528

**Reply by brianhuster** [reply]
It's just I don't want, it doesn't mean I will try to prevent people from doing so. It's just I won't do something to others that I don't want to be done. 

Anyway, it's just my suggestion, I won't try to put any pressure.
